woocommerce/woocommerce
# Code refactor on marketing components

### All Submissions:


### Changes proposed in this Pull Request:

This PR contains code refactor on marketing components:

- `CardHeaderTitle` component is introduced so that it can be used outside of `CollapsibleCard` component.
- `CenteredSpinner` component is introduced so that we can easily use it anywhere without having to repeat CSS for it.
- There were two `Plugin` types. They are renamed to `InstalledPlugin` and `RecommendedPlugin` to make them clearer and avoid confusion.
- `PluginCardBody` is modified to accept `className` props, and its `description` prop type is changed from `string` type to `ReactNode` type to make it more flexible and to be used in later PRs.

The changes introduced here would be used in subsequent PRs for new features in Multichannel Marketing page. It should not break existing functionality in the page.

### How to test the changes in this Pull Request:

Pre-condition: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`).

Test instructions:

1. Go to Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. Click around in the page and things should be working as expected, e.g.:
    - One-click installation should work.
    - Clicking on tabs should work.
    - Clicking on buttons should work as expected.

### Other information:




### FOR PR REVIEWER ONLY:
